main.go

```diff
@@ -171,16 +158,28 @@ func runController(ctx context.Context, kubeConfig *rest.Config) error {
 		).
 		WithGetValuesFuncs(addonConfigValuesFn, addonhelm.GetValuesFunc(k8sClient)).
 		WithAgentRegistrationOption(registrationOption).
-		WithScheme(scheme.Scheme).
+		WithScheme(scheme).
 		BuildHelmAgentAddon()
 	if err != nil {
-		klog.Errorf("failed to build agent %v", err)
+		logger.Error(err, "failed to build agent")
 		return err
 	}
 
 	err = mgr.AddAgent(mcoaAgentAddon)
 	if err != nil {
-		klog.Fatal(err)
+		logger.Error(err, "unable to add agent")
+		return err
+	}
+
+	disableReconciliation := os.Getenv("DISABLE_RECONCILIATION")
```

Comment:
Maybe this can be a CLI option. I think this can be interesting to do for testing reasons.

Reply:
Did you run into a use-case where you needed this feature? If we keep it I would change it to `DISABLE_WATCHER_CONTROLLER`. I'm fine with moving it to CLI option if we want to keep it